platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -13,204 +13,259 @@
    *
-   * <p>[Y, Y] (4) Felix file prop did not change, do nothing. This is the happy path.
+   * <p>[Y, null] [Y, X] (3) Felix prop was externally changed or removed. This is an error.
    *
-   * @param event the configuration event for which an update is needed
+   * <p>[Y, Y] (4) Felix file prop did not change, do nothing. This is the happy path. Properties
+   * will be updated on disk if necessary.
    */
-  private void handleUpdate(ConfigurationEvent event) {
-    Configuration configuration = getConfiguration(event);
-    if (configuration == null) {
-      LOGGER.debug("Configuration was null for pid {}", event.getPid());
-      return;
-    }
+  @Override
+  public void handleStore(ConfigurationContext context) throws IOException {
+    String pid = context.getServicePid();
+    File fileFromConfig = context.getConfigFile();
 
-    FelixConfig felixConfig = new FelixConfig(configuration);
-    String pid = felixConfig.getPid();
-    File felixFileNew = felixConfig.getFelixFile();
-    File felixFileOriginal = pidFileMap.get(pid);
+    CachedConfigData cachedConfigData = pidDataMap.get(pid);
+    File fileFromCache = (cachedConfigData != null) ? cachedConfigData.getFelixFile() : null;
 
-    // Config not tracked and will not be tracked? (1)
-    if (felixFileOriginal == null && felixFileNew == null) {
+    if (fileFromCache == null && fileFromConfig == null) {
+      // This config doesn't have an etc file, so we ignore this case (1)
       LOGGER.debug("Was not tracked and will not track pid {}", pid);
       return;
     }
 
-    boolean filePropChanged = felixConfig.filePropChanged(felixFileOriginal);
-
-    // Felix file prop did not change? (4)
-    if (filePropChanged) {
-      // Are we seeing the config for the first time? (2) (3)
-      if (felixFileOriginal != null) {
-        try {
-          felixConfig.setFelixFile(felixFileOriginal);
-        } catch (IOException e) {
-          LOGGER.error("Could not set felix file name, error writing to config admin: ", e);
-        }
-        return;
-      }
-      pidFileMap.put(pid, felixFileNew);
+    if (fileFromCache == null) {
+      // An etc config file was just dropped and we're seeing it for the first time (2)
       LOGGER.debug("Tracking pid {}", pid);
+      CachedConfigData createdConfigData = new CachedConfigData(context);
+      pidDataMap.put(pid, createdConfigData);
+      writeIfNecessary(context, createdConfigData);
+      return;
+    }
+
+    if (!Objects.equals(fileFromConfig, fileFromCache)) {
+      // The felix file prop changed, which is not allowed (3)
+      String msg =
+          (fileFromConfig == null)
+              ? format("Felix filename has been illegally removed, was [%s]", fileFromCache)
+              : format(
+                  "Felix filename has been illegally changed from [%s] to [%s]",
+                  fileFromCache, fileFromConfig);
+      throw new IllegalStateException(msg);
```

Comment:
❗️ Have you run itests against this change? I remember that when attempting etc sync last time, itests failed because they were touching this case. And we decided that, if this happens, we update the in-memory config with the old felix.fileinstall property and leave the property alone on disk.